src/Sulu/Bundle/LocationBundle/Resources/js/containers/Location/LocationOverlay.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+import React from 'react';
+import {action, observable, reaction, toJS} from 'mobx';
+import {observer} from 'mobx-react';
+import {Form, Input, Number} from 'sulu-admin-bundle/components';
+import Overlay from 'sulu-admin-bundle/components/Overlay';
+import {translate} from 'sulu-admin-bundle/utils/Translator';
+import {Map, Marker, TileLayer} from 'react-leaflet';
+import {SingleAutoComplete} from 'sulu-admin-bundle/containers';
+import type {Location as LocationValue} from '../../types';
+import locationOverlayStyles from './locationOverlay.scss';
+
+type Props = {
+    initialValue: ?LocationValue,
+    onClose: () => void,
+    onConfirm: (?LocationValue) => void,
+    open: boolean,
+};
+
+@observer
+class LocationOverlay extends React.Component<Props> {
+    @observable mapLat: number;
+    @observable mapLong: number;
+    @observable mapZoom: number;
+
+    @observable markerLat: ?number;
+    @observable markerLong: ?number;
+
+    @observable title: ?string;
+    @observable street: ?string;
+    @observable number: ?string;
+    @observable code: ?string;
+    @observable town: ?string;
+    @observable country: ?string;
+
+    updateInitialDataDisposer: () => *;
```

Comment:
use `() => void` (nothing happens with the return value, right?)

Reply:
unfortunately mobx declares the return type of `reaction` as `() => mixed`, therefore i used the same type as in other parts of the sulu code 🙂  https://github.com/mobxjs/mobx/blob/mobx4-master/flow-typed/mobx.js#L326